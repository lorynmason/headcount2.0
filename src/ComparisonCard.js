import React from 'react'
import './comparedCard.css'

const ComparisonCard = ({appState, resetComparison}) => {
  const comparedAvg = Object.keys(appState.compareCard)
  const firstAvg = comparedAvg[0]+ ': ' + appState.compareCard[comparedAvg[0]]
  const secondAvg = comparedAvg[1]+ ': ' + appState.compareCard[comparedAvg[1]]
  const arrowLeft = '<---- '
  const arrowRight = ' ---->'
  
  return (
  <div className="cards comparedCard clickedcard">
    <button className="tooltip" onClick={()=>resetComparison()}><span className="tooltiptext">click to reset comparison</span>X</button>
    <h3>
      {firstAvg} 
    </h3>
    <h3>
       {arrowLeft} {appState.compareCard.compared} {arrowRight}
    </h3>
    <h3>
      {secondAvg}
    </h3>
  </div>
)
}
  
export default ComparisonCard