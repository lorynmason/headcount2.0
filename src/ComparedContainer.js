import React from 'react';
import Card from './Card.js';
import './styles/main.scss';
import ComparisonCard from './ComparisonCard.js'


const ComparedContainer = ({appState, resetComparison, displaySelected}) => {
  let card1;
  let card2;
  let card3;
  if(appState.compared1 !== null) {
    card1 = <Card key={appState.compared1.location}
                  cardInfo={appState.compared1} 
                  displaySelected={displaySelected}/> 
  } 
  if(appState.compared2 !== null) {
    card2 = <Card key={appState.compared2.location} 
                  cardInfo={appState.compared2} 
                  displaySelected={displaySelected}/> 
  } 
  if(appState.compared1 !== null && appState.compared2 !== null) {
    card3 = <ComparisonCard appState={appState} resetComparison={resetComparison}/> 
  }
  
  return( 
    <div className="comparedCardContainer">
    <div>{card1}</div>
    <div>{card3}</div>
    <div>{card2}</div>
    </div>
  )
}

export default ComparedContainer