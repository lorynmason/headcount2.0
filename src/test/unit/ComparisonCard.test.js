import ComparisonCard from '../../../src/ComparisonCard.js';
import ComparedContainer from '../../../src/ComparedContainer.js';
import {shallow, mount} from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';


describe('ComparedCard', () => {
  let wrapper;
  let card1;
  let card2;
  let card3;
  let init;
  let resetComparison;
  beforeEach(() => {
    resetComparison = jest.fn()
    card1 = {
      location: "ADAMS COUNTY 14",
      selected: false,
      stats:
      {2004: 0.228,
      2005: 0.3,
      2006: 0.293,
      2007: 0.306,
      2008: 0.673,
      2009: 1,
      2010: 1,
      2011: 1,
      2012: 1,
      2013: 0.998,
      2014: 1}
      }
      card2 = {
        location: "ADAMS COUNTY 14",
        selected: false,
        stats:
        {2004: 0.228,
        2005: 0.3,
        2006: 0.293,
        2007: 0.306,
        2008: 0.673,
        2009: 1,
        2010: 1,
        2011: 1,
        2012: 1,
        2013: 0.998,
        2014: 1}
        }
        card3 = {'ADAMS COUNTY 14': 0.709,
        COLORADO: 0.53,
        compared: 0.748}
        
        init = {
          compared1: card1,
          compared2: card2,
          compareCard: card3
        }

    wrapper = shallow(
      <ComparisonCard appState={init} resetComparison={resetComparison}/>
    ); 

  })
  it('should match the snapshot with all data passed  in correctly', () => {
    
    expect(wrapper).toMatchSnapshot();
  })

  it('should call resetComparison on button click', () => {
    const button = wrapper.find('button')
    

    button.simulate('click')

    expect(resetComparison).toHaveBeenCalled()

  })
})