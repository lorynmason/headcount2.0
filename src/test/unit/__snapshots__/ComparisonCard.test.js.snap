// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparedCard should match the snapshot with all data passed  in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    appState={
        Object {
            "compareCard": Object {
              "ADAMS COUNTY 14": 0.709,
              "COLORADO": 0.53,
              "compared": 0.748,
            },
            "compared1": Object {
              "location": "ADAMS COUNTY 14",
              "selected": false,
              "stats": Object {
                "2004": 0.228,
                "2005": 0.3,
                "2006": 0.293,
                "2007": 0.306,
                "2008": 0.673,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 0.998,
                "2014": 1,
              },
            },
            "compared2": Object {
              "location": "ADAMS COUNTY 14",
              "selected": false,
              "stats": Object {
                "2004": 0.228,
                "2005": 0.3,
                "2006": 0.293,
                "2007": 0.306,
                "2008": 0.673,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 0.998,
                "2014": 1,
              },
            },
          }
    }
    resetComparison={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="tooltip"
          onClick={[Function]}
>
          <span
                    className="tooltiptext"
          >
                    click to reset comparison
          </span>
          X
</button>,
        <h3>
          ADAMS COUNTY 14: 0.709
</h3>,
        <h3>
          &lt;---- 
           
          0.748
           
           ----&gt;
</h3>,
        <h3>
          COLORADO: 0.53
</h3>,
      ],
      "className": "cards comparedCard clickedcard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="tooltiptext"
>
              click to reset comparison
</span>,
            "X",
          ],
          "className": "tooltip",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "click to reset comparison",
              "className": "tooltiptext",
            },
            "ref": null,
            "rendered": "click to reset comparison",
            "type": "span",
          },
          "X",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ADAMS COUNTY 14: 0.709",
        },
        "ref": null,
        "rendered": "ADAMS COUNTY 14: 0.709",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "<---- ",
            " ",
            0.748,
            " ",
            " ---->",
          ],
        },
        "ref": null,
        "rendered": Array [
          "<---- ",
          " ",
          0.748,
          " ",
          " ---->",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO: 0.53",
        },
        "ref": null,
        "rendered": "COLORADO: 0.53",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="tooltip"
            onClick={[Function]}
>
            <span
                        className="tooltiptext"
            >
                        click to reset comparison
            </span>
            X
</button>,
          <h3>
            ADAMS COUNTY 14: 0.709
</h3>,
          <h3>
            &lt;---- 
             
            0.748
             
             ----&gt;
</h3>,
          <h3>
            COLORADO: 0.53
</h3>,
        ],
        "className": "cards comparedCard clickedcard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="tooltiptext"
>
                click to reset comparison
</span>,
              "X",
            ],
            "className": "tooltip",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "click to reset comparison",
                "className": "tooltiptext",
              },
              "ref": null,
              "rendered": "click to reset comparison",
              "type": "span",
            },
            "X",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ADAMS COUNTY 14: 0.709",
          },
          "ref": null,
          "rendered": "ADAMS COUNTY 14: 0.709",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "<---- ",
              " ",
              0.748,
              " ",
              " ---->",
            ],
          },
          "ref": null,
          "rendered": Array [
            "<---- ",
            " ",
            0.748,
            " ",
            " ---->",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO: 0.53",
          },
          "ref": null,
          "rendered": "COLORADO: 0.53",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
